File: code/backend/src/main/java/com/blog/entity/Article.java
package com.blog.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "article")
public class Article {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(length = 200)
    private String title;
    
    @Column(unique = true, length = 200)
    private String slug;
    
    @Column(columnDefinition = "LONGTEXT")
    private String content;
    
    @Column(length = 500)
    private String summary;
    
    @Column(name = "cover_image", length = 500)
    private String coverImage;
    
    @Column(name = "category_id")
    private Long categoryId;
    
    @Column(nullable = false)
    private Integer status = 0;
    
    @Column(name = "view_count", nullable = false)
    private Integer viewCount = 0;
    
    @Column(name = "is_deleted", nullable = false)
    private Boolean deleted = false;
    
    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;
    
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
    
    @Column(name = "published_at")
    private LocalDateTime publishedAt;
}

File: code/frontend/src/types/article.ts
export interface IArticle {
  id: number
  title: string
  slug: string
  content?: string
  summary?: string
  coverImage?: string
  categoryId?: number
  status?: number
  viewCount?: number
  createdAt?: string
  updatedAt?: string
  publishedAt?: string
  category?: ICategory
  tags?: ITag[]
}

export interface IArticleListParams {
  page?: number
  size?: number
  categoryId?: number
  tagId?: number
  status?: number
  keyword?: string
}

export interface IArticleCreateRequest {
  title: string
  content: string
  summary?: string
  coverImage?: string
  categoryId: number
  tagIds?: number[]
  status: number
}

export interface IArticleUpdateRequest extends IArticleCreateRequest {
  id: number
}
